test/src/util.ts

```diff
@@ -60,11 +61,15 @@ export function transpileFile(filePath: string): string {
     return result.trim();
 }
 
-export function executeLua(luaStr: string, withLib = true): any {
+export function executeLua(luaStr: string, withLib = true, with51Backport = true): any {
     if (withLib) {
         luaStr = minimalTestLib + luaStr;
     }
 
+    if (with51Backport) {
```

Comment:
Can this backport stuff be removed since no 5.1 lua is executed?

Reply:
When luaTarget is 5.1 and a variable destructuring is performed, e.g.
```ts
declare function myFunc(): [number, string];
let [a, b] = myFunc();
```
It has to rersult in
```lua
local a,b=unpack(myFunc())
```
Because Lua 5.1 doesn't have `table.unpack`, which is generated by 5.2 and onwards.

The tests that run Lua code rely on [Fengari](https://github.com/fengari-lua/fengari) which only implements Lua 5.3 runtime. `unpack` (that was generated by luaTarget 5.1) is not defined in Lua 5.3 and thus has to be backported in order to be able to test Lua 5.1 code in Lua 5.3 VM.

Reply:
Which tests are executed in Fengari that are compiled with target 5.1?

Reply:
- [@Test("loop versions")](https://github.com/Janne252/TypescriptToLua/blob/3da4f7e96e53fadbe328a5cc36af1bc7a291dd91/test/unit/loops.spec.ts#L324)
- [ @Test("Bitop [5.1]")](https://github.com/Janne252/TypescriptToLua/blob/3da4f7e96e53fadbe328a5cc36af1bc7a291dd91/test/unit/expressions.spec.ts#L107)
- [@Test("Assignment destructuring [5.1]")](https://github.com/Janne252/TypescriptToLua/blob/3da4f7e96e53fadbe328a5cc36af1bc7a291dd91/test/unit/assignmentDestructuring.spec.ts#L11)

Reply:
Those are only transpiled, not executed.

Reply:
My bad. Total of 14 tests fail without the backport:

```
FAIL: map iterable constructor
FAIL: map iterable constructor map
FAIL: map clear
FAIL: map delete
FAIL: map entries
FAIL: map foreach
FAIL: map foreach keys
FAIL: map get
FAIL: map get missing
FAIL: map has
FAIL: map has null
FAIL: map keys
FAIL: map values
FAIL: map size
```

They all fail for the same reason:
```
Error: LUA ERROR: [string "function __TS__ArrayEvery(arr,callbackfn)..."]:268: attempt to call a nil value (global 'unpack')
```

Reply:
Hmm those are not transpiled as 5.1 either... Sounds like something is off here, I will investigate.

Reply:
It's caused by build_lualib.ts, the lualib is transpiled as 5.1. For example the generated map.lua uses unpack call:
````lua
function Map.constructor(self,other)
    self.items = {}
    self.size = 0
    if __TS__InstanceOf(other, Map) then
        self.size = other.size
        for _, __forOfValue0 in ipairs(other:entries()) do
        local key,value=unpack(__forOfValue0)
            do
                self.items[key] = value
            end
        end
    else
        if other~=nil then
            self.size = #other
            for _, __forOfValue1 in ipairs(other) do
            local key,value=unpack(__forOfValue1)
                do
                    self.items[key] = value
                end
            end
        end
    end
end
```

Reply:
Does changing the lua version to 5.3 in build_lualib.ts fix that? I am fine with you changing that for now so we can get rid of this backport stuff, since that is just treating symptoms instead of the actual cause.

We will have to come up with a better fix later, probably changing the map implementation so it does not use unpack (or any version-sensitive stuff) would be the best answer, but we can do that in a different PR.

Reply:
If the luaTarget for build_lualib.ts is changed to 5.3, then lualib would not work with Lua 5.1, as it would contain `table.unpack` (at the moment). If we don't want to use backports, the other option is to add a polyfill so that transpiled 5.1 code supports `table.unpack`, e.g.

```lua
if not table.unpack then
    function table.unpack(...)
        return unpack(...)
    end
end
```

This polyfill should probably be included in the lualib, but ideally it would only be emitted when the project luaLib target is set to 5.1.

Reply:
I acutally just hotfixed unpack out of the map implementation, if you merge master into your branch that should hopefully fix the failing tests.